test/audit/CollateralImbalance.t.sol

```diff
@@ -0,0 +1,254 @@
+        vm.assertEq(collateralBalance1, totalSupply1);
+        vm.assertEq(totalSupply1, 0);
+    }
+
+    function test_bridgeToCollateralRouter_CausesAccountingError() public {
+        bridgeToSynthetic();
+        uint256 collateralBalance0 = lsp7.balanceOf(address(collateralToken));
+        uint256 totalSupply0 = syntheticToken.totalSupply();
+
+        bridgeToken(
+            address(collateralToken).addressToBytes32(),
+            syntheticToken,
+            collateralToken,
+            DESTINATION,
+            ORIGIN,
+            localMailbox
+        );
+
+        uint256 collateralBalance1 = lsp7.balanceOf(address(collateralToken));
+        uint256 totalSupply1 = syntheticToken.totalSupply();
+
+        console.log("Previous Balance");
+        console.log("balanceOf(collateral)", collateralBalance0);
+        console.log("totalSupply()", totalSupply0);
+        console.log("Post Balance");
+        console.log("balanceOf(collateral)", collateralBalance1);
+        console.log("totalSupply()", totalSupply1);
+
+        // when bridging to the collateral token router itself,
+        // balance gets trapped in the collateral token router
+        vm.assertEq(collateralBalance1, totalSupply1);
+        vm.assertEq(totalSupply1, 0);
+    }
+
+    function test_bridgeToSynthetic_UnknownAddress() public {
+        uint256 collateralBalance0 = lsp7.balanceOf(address(collateralToken));
+        uint256 totalSupply0 = syntheticToken.totalSupply();
+
+        lsp7.mintTo(ATTACKER, amount);
+        vm.prank(ATTACKER);
+        lsp7.authorizeOperator(address(collateralToken), amount, "");
+
+        address NOONE = makeAddr("noop");
+        // bridgeToken(
+        //     NOONE.addressToBytes32(),
+        //     collateralToken,
+        //     syntheticToken,
+        //     ORIGIN,
+        //     DESTINATION,
+        //     remoteMailbox
+        // );
+        bridgeToken(NOONE.addressToBytes32(), collateralToken, syntheticToken, ORIGIN, DESTINATION, remoteMailbox);
+
+        uint256 collateralBalance1 = lsp7.balanceOf(address(collateralToken));
+        uint256 totalSupply1 = syntheticToken.totalSupply();
+        console.log("Previous Balance");
+        console.log("balanceOf(collateral)", collateralBalance0);
+        console.log("totalSupply()", totalSupply0);
+        console.log("Post Balance");
+        console.log("balanceOf(collateral)", collateralBalance1);
+        console.log("totalSupply()", totalSupply1);
+        // when bridging from collateral to synthetic the collateral balance
+        // should equal the synthetic totalSupply regardless of who the
+        // synthetic tokens are minted to
+        vm.assertEq(collateralBalance1, totalSupply1);
+        vm.assertEq(totalSupply1, amount);
+    }
+
+    function test_bridgeToSynthetic_SyntheticTokenAddress() public {
+        uint256 collateralBalance0 = lsp7.balanceOf(address(collateralToken));
```

Comment:
Maybe we can provide these two as parameters to the function to perform some fuzzing?